examples/player/simid_player.js

```diff
@@ -240,67 +255,111 @@ class SimidPlayer {
    * on top of the video content.
    */
-  displayNonLinearCreative_() {
-    const dimensions = this.getNonLinearDimensions();
-    
-    this.simidIframe_.style.height = dimensions.height;
-    this.simidIframe_.style.width = dimensions.width;
-    this.simidIframe_.style.left = `${dimensions.x}px`;
-    this.simidIframe_.style.top = `${dimensions.y}px`;
+  displayNonLinearCreative(incomingMessage = CreativeMessage.REQUEST_COLLAPSE) {
+    const newDimensions = this.getCreativeDimensions_();
 
-    this.simidIframe_.style.position = "absolute";
+    if (this.isLinearAd_) {
+      this.simidProtocol.reject(incomingMessage, "Cannot collapse linear ads");
+      console.log("Cannot collapse linear ads");
+    } else if (!this.isValidDimensions_(newDimensions)) {
+      this.simidProtocol.reject(incomingMessage, "Dimensions bigger than player");
+      console.log("Dimensions bigger than player");
+    } else {
+      this.setSimidIframeDimensions_(newDimensions);
+      this.simidIframe_.style.position = "absolute";
+
+      this.contentVideoElement_.play();
+      this.simidProtocol.resolve(incomingMessage);
+    }
   }
 
   /**
    * Allows users to request resizing of the creative
-   * @param {!Object} incomingMessage Message sent from the 
-   *   creative to the player
+   * @param {!Object} incomingMessage Message sent from the creative to the player
    */
   onRequestResize(incomingMessage) {
+
     if (this.isLinearAd_) {
       this.simidProtocol.reject(incomingMessage, "Cannot resize linear ad");
       console.log("Cannot resize linear ad");
-      return;
+    
+    } else if (!this.isValidDimensions_(incomingMessage.args.creativeDimensions)) { 
+      this.simidProtocol.reject(incomingMessage, "Dimensions bigger than player");
+      console.log("Dimensions bigger than player");
+    
+    } else {
+      this.setSimidIframeDimensions_(incomingMessage.args.creativeDimensions)
+      this.simidProtocol.resolve(incomingMessage);
     }
-
-    this.simidIframe_.style.height = incomingMessage.args.creativeDimensions['height'];
-    this.simidIframe_.style.width = incomingMessage.args.creativeDimensions['width'];
-    this.simidIframe_.style.left = incomingMessage.args.creativeDimensions['x'];
-    this.simidIframe_.style.top = incomingMessage.args.creativeDimensions['y'];
-
-    this.simidProtocol.resolve(incomingMessage);
   }
 
   /**
    * Initializes the SIMID creative with all data it needs.
    * @private
    */
   sendInitMessage_() {
-    const videoDimensions = this.getFullVideoDimensions(this.contentVideoElement_);
+    const videoDimensions = this.getFullDimensions_(this.contentVideoElement_);
     // Since the creative starts as hidden it will take on the
     // video element dimensions, so tell the ad about those dimensions.
-    const creativeDimensions = this.getFullVideoDimensions(this.contentVideoElement_);
+    const creativeDimensions = this.getFullDimensions_(this.contentVideoElement_);
```

Comment:
Oh, I'm glad we saw this here. For non linears the creative dimensions will need to be the amount of space the creative has.

For example if our non linear is {x:5, y:5, width:20, height:20} we would need to pass that in here instead of the video dimensions.

Reply:
Done